### PR TITLE
feat: add auth and private routes

### PR DESCRIPTION
**Pull request by Maianki**
### Files to be checked
- [src/App.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
- [src/components/Form/Input/Input.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-83e933f060abefe7a5fa219653b3f5bb2528667a88ea384d765a99955a2303cd)
- [src/components/Navbar/Navbar.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-547f8328c15a9a29000b7ef4690ba4f1039d148d5c208e5a48d37a2acd21982e)
- [src/components/RequireAuth/RequireAuth.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-1f65fdcef107cb810aa662d4e7e21a93ec1ef33d1aa2c1c1c64981dfaa103407)
- [src/components/RequireAuth/RestrictAuth.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-733290fe1a6f4136e0a00ebb5428c3930c4ea7c592efc216afe6f5f946f32a31)
- [src/context/auth-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-50d25f7adf348217c540867a8937453cd2a126a2abdf2681a9e4a838a51f9714)
- [src/context/videos-and-category-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-e908d307e43003c21fc1b5e293786084d5a800e13416f4278a8c13486a83afa9)
- [src/pages/Login/Login.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-da005ba6520f0f42b7e1052a931cea46e6703aceea8989e37fa93718ac033ad1)
- [src/pages/Signup/Signup.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-a981778bc61ac908a780e1c6ab71acef92e39ae1fe892d7093ad25d12d3efb3e)
- [src/reducers/auth-reducer.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-a981778bc61ac908a780e1c6ab71acef92e39ae1fe892d7093ad25d12d3efb3e)
- [src/utils/APIEndPoints.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-auth?expand=1#diff-a981778bc61ac908a780e1c6ab71acef92e39ae1fe892d7093ad25d12d3efb3e)

### Live preview
[click here for live preview](https://shuttle-motion-video-library-git-add-auth-maianki.vercel.app/)

### CODE REVIEW

**Review by Hamza-777 — APPROVED** [review]
Looks good and clean.

File: src/components/RequireAuth/RequireAuth.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+
+export function RequireAuth() {
+  const {
+    auth: { encodedToken, isAuthenticated },
+  } = useAuth();
+  const location = useLocation();
+
+  if (encodedToken && isAuthenticated) return <Outlet />;
+  return <Navigate to='/login' state={{ from: location }} replace></Navigate>;
```
Comment (sainath7878) [inline src/components/RequireAuth/RequireAuth.jsx]: suggestion: You could just give a self closing tag to Navigate

File: src/components/RequireAuth/RestrictAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+
+export function RestrictAuth() {
+  const {
+    auth: { encodedToken, isAuthenticated },
+  } = useAuth();
+  const location = useLocation();
+
+  if (encodedToken && isAuthenticated)
+    return <Navigate to='/' state={{ from: location }} replace></Navigate>;
```
Comment (sainath7878) [inline src/components/RequireAuth/RestrictAuth.jsx]: same here

File: src/components/RequireAuth/RequireAuth.jsx (hunk 1 above)
Comment (Maianki) [reply src/components/RequireAuth/RequireAuth.jsx]: sure will do it.

File: src/components/RequireAuth/RestrictAuth.jsx (hunk 2 above)
Comment (Maianki) [reply src/components/RequireAuth/RestrictAuth.jsx]: Sure will do it.

File: src/utils/APIEndPoints.js [hunk 3]
```diff
@@ -1,2 +1,4 @@
-export const videos_API = "/api/videos";
-export const categories_API = "/api/categories";
+export const VIDEOS_API = "/api/videos";
+export const CATEGORIES_API = "/api/categories";
+export const LOGIN_API = "/api/auth/login";
+export const SIGNUP_API = "/api/auth/signup";
```
Comment (sainath7878) [inline src/utils/APIEndPoints.js:4]: This is awesome!

**Review by sainath7878 — APPROVED** [review]
Everything looks good. Neat and clean code. Great Work!

**Comment by Maianki** [comment]
Thanks for reviewing @Hamza-777 and @sainath7878. Done the changes merging PR.

**Pull request merged**